### ISSUE TITLE
Feat: allow custom & chained message preprocs

### ISSUE THREAD

**Issue by smashah** [open-wa/wa-automate-nodejs#2931]

**Comment by smashah** [comment]
@github-actions run

<details>
<summary>⚡ Release!  ⚡</summary>

```js
(async () => {
function exec(cmd) {
  console.log(execSync(cmd).toString());
}
//set the version type
process.env.VERS = "minor"

// Config
const gitUserEmail = "github-actions[bot]@users.noreply.github.com";
const gitUserName = "github-actions[bot]";

exec(`echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" > .npmrc`);
exec(`git config --global user.email "${gitUserEmail}"`);
exec(`git config --global user.name "${gitUserName}"`);
exec(`npm i -D`);
exec(`npm run release-ci $VERS`);

// types only package
exec('npm config set //registry.npmjs.org/:_authToken $NPM_TOKEN')
exec(`cd ./types-only && npm version $VERS && npm run build && npm publish && cd ..`);
exec(`git commit -a -m 'updated types-only package'`);
exec(`git push --force`);

//comment on the issue
var result = execSync(`npx auto-changelog -o ./tempchangelog.txt --commit-limit false --template ./compact-keepachangelog.hbs --stdout`).toString();

    await postComment(result);

//create changelog image
exec(`npm run release-image`);
exec(`git commit -a -m 'updated release-image'`);
exec(`git push --force`);
  })();
```
</details>

**Comment by smashah** [comment]
# Changelog

## 🚀 Release [4.48.0](https://github.com/open-wa/wa-automate-nodejs/compare/4.47.6...4.48.0) (2022-10-27)

- 📝 improved documentation [`4306e8a`](https://github.com/open-wa/wa-automate-nodejs/commit/4306e8a2a39d33a9867df29d7a8b3f3375f295cf)
- ✨ feat: add custom + chainable preprocs #2931 [`9e46c22`](https://github.com/open-wa/wa-automate-nodejs/commit/9e46c22f8345792493664d98b68d413dd3c10f76)
- 📝 update docs [`f4ed9c5`](https://github.com/open-wa/wa-automate-nodejs/commit/f4ed9c5ab68e63ba6708ed6cfb46ebcf35b52579)
- Release 4.48.0 [`d2c88d4`](https://github.com/open-wa/wa-automate-nodejs/commit/d2c88d420eb49d30c95ee73df742cc846e77a598)
- 💚 fix ci build issue with preprocs #2931 [`14acbc2`](https://github.com/open-wa/wa-automate-nodejs/commit/14acbc24a55193895c846f6921c5fabd4299fd70)
- 🥚 patches update [`584cddb`](https://github.com/open-wa/wa-automate-nodejs/commit/584cddb104f1b5d8f8521c6e3cc844d9202d7f27)
- 🥚 patches update [`2f0cfa8`](https://github.com/open-wa/wa-automate-nodejs/commit/2f0cfa858c1285dbbe25b85da0b311d78f0c0b21)
- updated types-only package [`5c5cc3d`](https://github.com/open-wa/wa-automate-nodejs/commit/5c5cc3dc7c31c26a2aaaf43d022b4ecf30e985ee)
